README.md

```diff
@@ -2,51 +2,113 @@
 
 Add footnotes functionality to your Wagtail project.
 
-## Usage
- - Add the app to `INSTALLED_APPS`:
-   ```python
-   INSTALLED_APPS = [
-       ...
-       "wagtail_footnotes",
-       ...
-   ]
-   ```
- - Add the footnotes `urls.py` to your project's `urls.py`:
-   ```python
-   from wagtail_footnotes import urls as footnotes_urls
-   urlpatterns = [
-       ...
-       path("footnotes/", include(footnotes_urls)),
-       ...
-   ]
-   ```
-   Note: The URL has to be defined as above as it is currently hardcoded in the Javascript.
- - Update your page models to show the footnotes field:
-   ```python
-   class InformationPage(BasePage):
+## ⚡ Quick start
+
+Add `wagtail_footnotes` to `INSTALLED_APPS`:
+
+```python
+INSTALLED_APPS = [
+    ...
+    "wagtail_footnotes",
+    ...
+]
+```
+
+Add the footnotes `urls.py` to your project's `urls.py`:
+
+```python
+from wagtail_footnotes import urls as footnotes_urls
+urlpatterns = [
+    ...
+    path("footnotes/", include(footnotes_urls)),
+    ...
+]
+```
+
+*Note: The URL has to be defined as above as it is currently hardcoded in the Javascript.*
+
+Update your page models to show the footnotes panel:
+
+```python
+class InformationPage(Page):
+    ...
+    content_panels = [
         ...
-        content_panels = [
-            ...
-            InlinePanel("footnotes", label="Footnotes"),
-        ]
-   ```
- - Update your `RichTextBlock`s 
-    - Add `"footnotes"` to the `features` arg for each `RichTextBlock` that you want to have this functionality
-    - You will also need to change any `RichTextBlock`s to `wagtail_footnotes.blocks.RichTextBlockWithFootnotes`
- - Update your page templates to include `{% include "wagtail_footnotes/includes/footnotes.html" %}`
- - Make and run migrations:
-   ```
-   ./manage.py makemigrations
-   ./manage.py migrate
-   ```
-
-## Settings
+        InlinePanel("footnotes", label="Footnotes"),
+    ]
+```
+
+### Showing Footnotes in page templates
+
+Update your page templates to include `{% include "wagtail_footnotes/includes/footnotes.html" %}`. You can copy from this template and instead include your own customized version.
+
+### Using footnotes in RichTextField
+
+Update any `RichTextField`s that you want to add footnotes feature. Add `"footnotes"` to the `features` arg for each `RichTextField` that you want to have this functionality. For instance:
+
+```
+class MyPage(Page):
```

Comment:
Hi @benjaoming Did you test out this usage here?

It doesn't work for me, I also tried it on earlier versions. I don't think it's a use case thats catered for in the package.

RichText field does work to a point (in the admin) but the rendered template doesn't receive any footnotes in the page context.

Reply:
I think it's what follows from https://github.com/torchbox/wagtail-footnotes/issues/35 - I don't remember this _not_ working.

I did this in my implementation:

```

{% richtext_footnotes page.get_body %}

{% include "wagtail_footnotes/includes/footnotes.html" %}
```

My `richtext_footnotes` template tag looks like this:

```
@register.simple_tag(takes_context=True)
def richtext_footnotes(context, html):
    """
    example: {% richtext_footnotes page.body|richtext %}

    html: already processed richtext field html
    Assumes "page" in context.
    """
    FIND_FOOTNOTE_TAG = re.compile(r'<footnote id="(.*?)">.*?</footnote>')

    if not isinstance(context.get("page"), Page):
        return html

    page = context["page"]
    if not hasattr(page, "footnotes_list"):
        page.footnotes_list = []
    footnotes = {str(footnote.uuid): footnote for footnote in page.footnotes.all()}

    def replace_tag(match):
        try:
            index = process_footnote(match.group(1), page)
        except (KeyError, ValidationError):
            return ""
        else:
            return f'<a href="#footnote-{index}" id="footnote-source-{index}"><sup>[{index}]</sup></a>'

    def process_footnote(footnote_id, page):
        footnote = footnotes[footnote_id]
        if footnote not in page.footnotes_list:
            page.footnotes_list.append(footnote)
        # Add 1 to the index as footnotes are indexed starting at 1 not 0.
        return page.footnotes_list.index(footnote) + 1

    return mark_safe(FIND_FOOTNOTE_TAG.sub(replace_tag, html))
```

Reply:
Btw, you're right that it doesn't work for the "frontend" - I think I realized this after editing the README. But you're welcome to adapt the above.

I think that I wanted to have some kind of template filter like the above included in the extension. I just dumped some code above, but you could also call the template filter like this:

```
{{ page.body_richtext|richtext|richtext_footnotes %}
```

Reply:
Yes thats right, the admin works but not the frontend.

I'd agree it seems like a worth while enhancement if you fancy doing it 😄